pkg/lib/scoped/token_retriever.go

```diff
@@ -82,6 +82,12 @@ func getAPISecret(logger logrus.FieldLogger, kubeclient operatorclient.ClientInt
 func filterSecretsBySAName(saName string, secrets *corev1.SecretList) map[string]*corev1.Secret {
 	secretMap := make(map[string]*corev1.Secret)
 	for _, ref := range secrets.Items {
+		// Avoid referencing the "ref" created by the range-for loop as
+		// the secret stored in the map will change if there are more
+		// entries in the list of secrets that the range-for loop is
+		// iterating over.
+		ref := ref
```

Comment:
I really prefer not forcing unintuitive inner/outer variable collisions to avoid this.  
I'll LGTM, but IMHO the better approach is to use a non-colliding loop-scoped variable with a note.  
I concede that this yields a better code delta, but if it requires a substantial comment like the one provided, is it really an improvement?

Reply:
This has already merged, but I 100% agree with the sentiment.

In this case, @joelanford pointed out that this is the canonical solution that he usually sees to address this intuitive problem. As you said, we could create our own way of avoiding the issue and forego the comment, but I would rather:
- Adhere to standards found in multiple go projects.
- Not "hide" the problem by coming up with our own solution.
- Document the reason for the approach with a comment so our team is aware of this pattern and can apply it when writing/reviewing code later.

Again, I don't know that I would have supported this approach had I been there when the pattern was created, but now that it is established I will respect it.